### ISSUE TITLE
Replace version.txt with git tags and/or github releases

### ISSUE THREAD

**Issue by xrotwang** [CHIELDOnline/CHIELD#153]
I'm not exactly sure what the `version.txt` file is supposed to do, but it looks as if it would be simpler or at least more transparent to use git tags and/or github releases for named snapshots of the repository. Making ZENODO [archive releases automatically](https://guides.github.com/activities/citable-code/) would also give you a DOI and some long-term preservation.

**Comment by seannyD** [comment]
Yes, you're right - full release versions should be dealt with through github.  This was meant to handle synchronisation between local cookie data and minor updates of the database.

When a user is editing data, the variables they create are stored in a cookie.  This is handy for them, because their data is not yet part of the database.  When the database is rebuilt, incorporating their changes, the "version number" changes.  The web program gets the "version number"from a table in the SQLite database, checks whether it is different from the version number in the cookie, and if so, clears their cookie list.  

The `version.txt` just contains the prefix for the version number that's stored in the database.  The string used by the web program is actually the git commit id.  The R code for building the SQLite table:

```
getVersion = function(){
  gitRevision = system("git rev-parse HEAD",intern = T)
  CHIELD.version = readLines("../version.txt")
  return(data.frame(version=CHIELD.version,
                    gitRevision = gitRevision))
}
```

The progam uses the `gitRevision` field, so actually the "version.txt" file could be removed.  However, I'll have to check if it's used for anthing else.  I'll keep this issue open to remind me.

BTW - I really appreciate your review!

**Comment by xrotwang** [comment]
Ah, I see. So it's some sort of [cache busting](https://www.keycdn.com/support/what-is-cache-busting/). But it's also a bit half-assed, isn't it? If the user had "git in the browser", the javascript could actually determine from the commit hash, whether the new database actually does "incorporate their changes".

This is actually part of why I'm not fully convinced of your data editing solution: Hiding the git/github complexity from the user seems to create problems of its own. Considering that most of the app's functionality is pure javascript, a better option could be: Have user's run the app locally in their browser, serving data from their local fork of the repository. Of course this would mean users have to
- edit the data in their fork
- recreate the sqlite db before being able to look at the new data

But introducing users to this workflow may be more maintainable in the long term.

**Comment by seannyD** [comment]
Yes, it is a kind of cache busting, but only for a very minor feature: placing recently entered variables in your suggestion list in the gap before they're officially incorporated into the database.  This is just to try and limit people creating lots of different variable names for the same concept.  I was imagining the app as just a fancy way of creating files for the github repository, with some extra features to actually create a pull request directly.  The data will never be immediately incorporated with any solution, because I want to review each submission as it comes in.

As far as possible, I want to avoid people having to interact with github/ serving data.  At the moment, adding data is quite intuitive and I'd like this to be accessible to e.g. Masters-level students.  Of course, people can contribute directly through git if they want.
